src/main/java/ManageInputs.java

```diff
@@ -0,0 +1,209 @@
+                throw new UnexpectedCommandException();
+            }
+            System.out.println("event timeline not specified");
+            throw new UnexpectedCommandException();
+        }
+        try {//description not specified
+            String description = line.substring(6, indexFrom - 1);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("event description not specified");
+            throw new UnexpectedCommandException();
+        }
+        String from = line.substring(indexFrom + 5, indexTo - 1);
+        String to = line.substring(indexTo + 2);
+        String description = line.substring(5, indexFrom - 1);
+
+        tasks.add(index, new Event(description, from, to));
+    }
+
+    public static void dealWithDeadline(ArrayList<Task> tasks, int index, String line) throws UnexpectedCommandException, IOException {
+        int indexBy = line.indexOf("by");
+        int space = line.indexOf(" ");
+        if (indexBy == -1) {//invalid format
+            System.out.println("Invalid format! Enter deadline in the format: deadline (description) by (deadline)");
+            throw new UnexpectedCommandException();
+        }
+        try {//deadline / both not specified
+            String by = line.substring(indexBy + 2);
+        } catch (IndexOutOfBoundsException e) {
+            try {
+                String description = line.substring(space, indexBy - 1);
+            } catch (IndexOutOfBoundsException f) {
+                System.out.println("deadline description and deadline not specified");
+                throw new UnexpectedCommandException();
+            }
+            System.out.println("deadline not specified");
+            throw new UnexpectedCommandException();
+        }
+        try {//deadline not specified
+            String description = line.substring(space, indexBy - 1);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("deadline description not specified");
+            throw new UnexpectedCommandException();
+        }
+        String description = line.substring(space, indexBy - 1);
+        String by = line.substring(indexBy + 2);
+
+        tasks.add(index, new Deadline(description, by));
+    }
+
+    public static void dealWithTodo(ArrayList<Task> tasks, int index, String line) throws UnexpectedCommandException, IOException {
+        int indexSpace = line.indexOf(" ");
+        if (indexSpace == -1) {
+            System.out.println("todo description not specified");
+            throw new UnexpectedCommandException();
+        }
+
+        String description = line.substring(indexSpace);
+
+        tasks.add(index, new Todo(description));
+
+    }
+
+    private void handleUnexpectedCommand(boolean isValidCommand) throws UnexpectedCommandException {
+        if (!isValidCommand) {
+            throw new UnexpectedCommandException();
+        }
+    }
```

Comment:
I'm not sure what this abstraction is supposed to do. Is this necessary? If a command is invalid, is it possible to just `throw` the exception straight?